undefined tunny function

when attempting to build with `go1.12.14` I get 

    pre_commit.go:102:15: undefined: tunny.CreatePoolGeneric

It should be noted that while this function does exist in the version of tunny specified by glide it no longer exists in the current version of tunny. I'm not suggesting that's the bug (it does exist in what glide pulled down). I'm suggesting that _if_ you're going to poke it anyway, you may want to update to the latest tunny API.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: go
 go:
-  - '1.6'
-  - '1.7'
+  - '1.11.9'
+  - '1.12.4'
 
 before_install:
   # Setup Glide

--- a/glide.lock
+++ b/glide.lock
@@ -1,10 +1,10 @@
-hash: c955d5292bc09953440fb8c69fa258118edb0ca272262682d216086fb8dcdcf0
-updated: 2016-09-08T23:48:33.490565716-07:00
+hash: f325e1261961346daec92f823d90e12c47142279dc062cec551e61e2c021e5ea
+updated: 2019-04-22T18:15:24.19885-07:00
 imports:
 - name: github.com/fatih/color
   version: 87d4004f2ab62d0d255e0a38f1680aa534549fe3
 - name: github.com/jeffail/tunny
-  version: 18dedac4b5873be2596b7a49ff0159e6691f9836
+  version: 59cfa8fcb19f5acba8db07a0dd3cb2fa7edbc228
 - name: github.com/mattn/go-colorable
   version: ed8eb9e318d7a84ce5915b495b7d35e0cfe7b5a8
 - name: github.com/mattn/go-isatty

--- a/glide.yaml
+++ b/glide.yaml
@@ -3,6 +3,6 @@ import:
 - package: github.com/urfave/cli
   version: 1efa31f08b9333f1bd4882d61f9d668a70cd902e # v1.18.0
 - package: github.com/jeffail/tunny
-  version: 18dedac4b5873be2596b7a49ff0159e6691f9836
+  version: 0.1.1
 - package: github.com/fatih/color
   version: v1.0.0

--- a/hooks/pre_commit.go
+++ b/hooks/pre_commit.go
@@ -91,35 +91,21 @@ type Result struct {
 }
 
 // Uses a pool sized to the number of CPUs to run all the executables. It's
-// sized to the CPU count so that we fully utilized the hardwire but don't
+// sized to the CPU count so that we fully utilized the hardware but don't
 // context switch in the OS too much.
 func runExecutablesInParallel(executables []*context.Executable, files []string) []*Result {
 	bufferSize := len(executables)
-
-	in := make(chan *context.Executable, bufferSize)
 	out := make(chan *Result, bufferSize)
 
-	pool, err := tunny.CreatePoolGeneric(runtime.NumCPU()).Open()
-	if err != nil {
-		panic(err)
-	}
-
+	pool := tunny.NewFunc(runtime.NumCPU(), func(executable interface{}) interface{} {
+		return runPreCommitExecutable(executable.(*context.Executable), files)
+	})
 	defer pool.Close()
 
-	for _, executable := range executables {
-		in <- executable
-
+	for index := range executables {
+		executable := executables[index]
 		go func() {
-			_, err := pool.SendWork(func() {
-				executable := <-in
-
-				out <- runPreCommitExecutable(executable, files)
-			})
-
-			// Something real bad happened
-			if err != nil {
-				panic(err)
-			}
+			out <- pool.Process(executable).(*Result)
 		}()
 	}
 

